dirty_cat/_super_vectorizer.py

```diff
@@ -427,6 +429,15 @@ def _apply_cast(self, X: pd.DataFrame) -> pd.DataFrame:
         for col in self.imputed_columns_:
             X.loc[:, col] = _replace_missing_in_cat_col(X[col])
         for col, dtype in self.types_.items():
+            # if categorical, add the new categories to prevent
+            # them to be encoded as nan
+            if pd.api.types.is_categorical_dtype(dtype):
+                knwon_categories = dtype.categories
+                new_categories = pd.unique(X[col])
+                dtype = pd.CategoricalDtype(
+                    categories=knwon_categories.union(new_categories)
```

Comment:
```suggestion
                known_categories = dtype.categories
                new_categories = pd.unique(X[col])
                dtype = pd.CategoricalDtype(
                    categories=known_categories.union(new_categories)
```
I guess is what you meant